lib/flags.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const { Uint64 } = require('./uint64');
+
+class Flags {
+  static from(...args) {
+    const values = new Map(args.map((v, i) => [v, i]));
+    class FlagsClass {
+      static from(...args) {
+        return new FlagsClass(...args);
+      }
+
+      constructor(...args) {
+        if (args[0] instanceof Uint64) {
+          this.value = new Uint64(args[0]);
+        } else {
+          this.value = new Uint64(0);
+          args.forEach(arg => this.set(arg));
+        }
+      }
+
+      has(key) {
+        return values.has(key);
+      }
+
+      get(key) {
+        return values.has(key) ?
+          Uint64.shiftRight(this.value, values.get(key)).toUint32() & 1 : -1;
+      }
+
+      set(key) {
+        if (values.has(key)) {
+          this.value.or(new Uint64(1).shiftLeft(values.get(key)));
+        }
+      }
+
+      unset(key) {
+        if (values.has(key)) {
```

Comment:
So if the user provided invalid value we will just silently ignore and proceed?
e.g. we have flags of ('Uno', 'Due', 'Tre') and user writes `flags.toggle('Duo')` which will just ignore the call without any error.

I think all such cases should result in TypeError.
/cc @belochub @nechaido @tshemsedinov

Reply:
Ditto https://github.com/metarhia/common/pull/201/files#r232201310.